packages/node_modules/@webex/internal-plugin-conversation/test/unit/spec/conversation.js

```diff
@@ -145,20 +145,78 @@ describe('plugin-conversation', () => {
+        .then(() => {
+          assert.called(webex.request);
+        }));
+
+      it('should not retrieve non reaction summary objects', () => webex.internal.conversation.getReactionSummaryByParentId(convoUrl, 'test-id')
+        .then((result) => {
+          assert.equal(result.length, 3);
+          assert.notInclude(result, {type: 'notAReaction'});
+        }));
+    });
+
+    describe('#listAllChildActivitiesByParentId()', () => {
+      let dayCount = 0;
+      const createActivityItemBatch = (itemCount) => {
+        const counter = [...Array(itemCount).keys()];
+
+        dayCount += 1;
+
+        return counter.map((n) => ({
+          published: new Date(2020, 0, 1, dayCount, n)
+        }));
+      };
+
+      const createMockChildResponse = (itemsToReturn = 10) => {
+        const response = {
+          body: {
+            items: createActivityItemBatch(itemsToReturn)
+          },
+          headers: {
+            link: '<https://www.cisco.com>; rel=next'
+          }
+        };
+
+        return response;
+      };
+
+      beforeEach(() => {
+        webex.request = sinon.stub()
+          .onCall(0)
+          .returns(Promise.resolve(createMockChildResponse()))
+          .onCall(1)
+          .returns(Promise.resolve(createMockChildResponse()))
+          .onCall(2)
+          .returns(Promise.resolve(createMockChildResponse(3)))
+          .returns(Promise.resolve(Promise.resolve({
+            body: {
+              items: []
+            },
+            headers: {}
+          })));
       });
+      it('retrieves correct children count', () => webex.internal.conversation.listAllChildActivitiesByParentId({conversationUrl: convoUrl, activityParentId: '123-abc'})
+        .then((res) => {
+          assert.equal(res.length, 23);
+        }));
 
-      it('should not retrieve non reaction summary objects', () => {
-        webex.internal.conversation.getReactionSummaryByParentId(convoUrl, 'test-id')
-          .then((result) => {
-            assert.equal(result.length, 3);
-            assert.notInclude(result, {type: 'notAReaction'});
+      it('calls #listChildActivitiesByParentId() to initiate the request', () => {
+        const spy = sinon.spy(webex.internal.conversation, 'listChildActivitiesByParentId');
+
+        return webex.internal.conversation.listAllChildActivitiesByParentId({conversationUrl: convoUrl, activityParentId: '123-abc'})
+          .then(() => {
+            assert(spy.calledOnce);
           });
       });
+
+      it('returns children in ascending published order', () => webex.internal.conversation.listAllChildActivitiesByParentId({conversationUrl: convoUrl, activityParentId: '123-abc'})
+        .then((res) => {
+          const firstMessageOlderThanLastMessage = res[0].published < res[res.length - 1].published;
+
+          assert.equal(firstMessageOlderThanLastMessage, true, 'activities out of order');
```

Comment:
Nitpick (not required to change but would be more consistent with other tests): use `assert.isTrue` instead